internal/receiver/databricksreceiver/metrics_provider.go

```diff
@@ -0,0 +1,104 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package databricksreceiver
+
+import (
+	"fmt"
+
+	"go.opentelemetry.io/collector/model/pdata"
+)
+
+const (
+	jobScheduleStatusMetric  = "databricks.jobs.schedule.status"
+	totalJobsMetric          = "databricks.jobs.total"
+	totalActiveJobsMetric    = "databricks.jobs.active.total"
+	taskScheduleStatusMetric = "databricks.tasks.schedule.status"
+)
+
+// metricsProvider wraps a databricksClientInterface and provides metrics for databricks
+// endpoints.
+type metricsProvider struct {
+	dbClient databricksClientInterface
+}
+
+func newMetricsProvider(dbClient databricksClientInterface) metricsProvider {
+	return metricsProvider{dbClient: dbClient}
+}
+
+func (p metricsProvider) addJobStatusMetrics(ms pdata.MetricSlice) ([]int, error) {
+	jobs, err := p.dbClient.jobs()
+	if err != nil {
+		return nil, fmt.Errorf("metricsProvider.addJobStatusMetrics(): %w", err)
+	}
+
+	addJobCountMetric(ms, int64(len(jobs)))
+
+	jobPts := mkGauge(ms, jobScheduleStatusMetric)
+	taskPts := mkGauge(ms, taskScheduleStatusMetric)
+
+	var jobIDs []int
+	for _, j := range jobs {
+		jobIDs = append(jobIDs, j.JobID)
+		jobPt := jobPts.AppendEmpty()
+		pauseStatus := pauseStatusToInt(j.Settings.Schedule.PauseStatus)
+		jobPt.SetIntVal(pauseStatus)
+		jobIDAttr := pdata.NewAttributeValueInt(int64(j.JobID))
+		jobPt.Attributes().Insert("job_id", jobIDAttr)
+		for _, task := range j.Settings.Tasks {
+			taskPt := taskPts.AppendEmpty()
+			taskPt.SetIntVal(pauseStatus)
+			taskAttrs := taskPt.Attributes()
+			taskAttrs.Insert("job_id", jobIDAttr)
+			taskAttrs.Insert("task_id", pdata.NewAttributeValueString(task.TaskKey))
+		}
+	}
+	return jobIDs, nil
```

Comment:
Are you planning to add the task and job properties in a future release?
Example: https://github.com/dloucasfx/signalfx-agent/commit/287456eabe2ed01ec351a878d1ceec16b1a0405d#diff-84aefb0f500a3fb8b2b091d108e98045485ce47b27c662fb3932b8161f716f4bR147-R162

Reply:
We don't have a concept of properties in otel so I punted on this. Do you think task type should be an attribute (which becomes a dimension)?

Reply:
oh, I was not aware that properties are not implemented in the SFX exporter, well, that's a bummer.
I think task type don't change frequently, so yes, we can add them as dimensions

Reply:
ftr these are available in otel https://github.com/signalfx/splunk-otel-collector/blob/1d289f68e9896eb58fdea3ef290e83a50774a5b0/internal/receiver/smartagentreceiver/dimension.go#L28. Considering the size of this PR already my 2c would be to add metadata exporter adoption in a later pass.

Reply:
Ok, sounds good, I'll add task type as an attribute. BTW I meant that there isn't a concept of properties in Otel (receivers are exporter agnostic), but there is some support for properties in the SFx exporter, but it's using a metadata exporter which is apparently experimental, and I'm not quite sure it's the right approach here. Can always do this later if we decide to.